ui/ducks/swaps/swaps.test.js

```diff
@@ -288,6 +459,234 @@ describe('Ducks - Swaps', () => {
+        state.metamask.swapsState.tokens,
+      );
+    });
+  });
+
+  describe('getSwapsWelcomeMessageSeenStatus', () => {
+    it('returns', () => {
+      const state = createSwapsMockStore();
+      expect(swaps.getSwapsWelcomeMessageSeenStatus(state)).toBe(
+        state.metamask.swapsState.swapsWelcomeMessageHasBeenShown,
+      );
+    });
+  });
+
+  describe('getTopQuote', () => {
+    it('returns a top quote', () => {
+      const state = createSwapsMockStore();
+      expect(swaps.getTopQuote(state)).toBe(
+        state.metamask.swapsState.quotes.TEST_AGG_BEST,
+      );
+    });
+  });
+
+  describe('getApproveTxId', () => {
+    it('returns approveTxId', () => {
+      const state = createSwapsMockStore();
+      expect(swaps.getApproveTxId(state)).toBe(
+        state.metamask.swapsState.approveTxId,
+      );
+    });
+  });
+
+  describe('getTradeTxId', () => {
+    it('returns tradeTxId', () => {
+      const state = createSwapsMockStore();
+      expect(swaps.getTradeTxId(state)).toBe(
+        state.metamask.swapsState.tradeTxId,
+      );
+    });
+  });
+
+  describe('getUsedQuote', () => {
+    it('returns selected quote', () => {
+      const state = createSwapsMockStore();
+      expect(swaps.getUsedQuote(state)).toMatchObject(
+        state.metamask.swapsState.quotes.TEST_AGG_2,
+      );
+    });
+
+    it('returns best quote', () => {
+      const state = createSwapsMockStore();
+      state.metamask.swapsState.selectedAggId = null;
+      expect(swaps.getUsedQuote(state)).toMatchObject(
+        state.metamask.swapsState.quotes.TEST_AGG_BEST,
+      );
+    });
+  });
+
+  describe('getDestinationTokenInfo', () => {
+    it('returns destinationTokenInfo', () => {
+      const state = createSwapsMockStore();
+      expect(swaps.getDestinationTokenInfo(state)).toBe(
+        state.metamask.swapsState.fetchParams.metaData.destinationTokenInfo,
+      );
+    });
+  });
+
+  describe('getUsedSwapsGasPrice', () => {
+    it('returns customGasPrice', () => {
+      const state = createSwapsMockStore();
+      state.metamask.swapsState.customGasPrice = 5;
+      expect(swaps.getUsedSwapsGasPrice(state)).toBe(
+        state.metamask.swapsState.customGasPrice,
+      );
+    });
+  });
+
+  describe('getApproveTxParams', () => {
+    it('returns approveTxParams', () => {
+      const state = createSwapsMockStore();
+      state.metamask.swapsState.quotes.TEST_AGG_2.approvalNeeded = {
+        data: '0x095ea7b300000000000000000000000095e6f48254609a6ee006f7d493c8e5fb97094cef0000000000000000000000000000000000000000004a817c7ffffffdabf41c00',
+        to: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
+        value: '0x0',
+        from: '0x2369267687A84ac7B494daE2f1542C40E37f4455',
+        gas: '0x12',
+        gasPrice: '0x34',
+      };
+      expect(swaps.getApproveTxParams(state)).toMatchObject({
+        ...state.metamask.swapsState.quotes.TEST_AGG_2.approvalNeeded,
+        gasPrice: 5,
```

Comment:
Why is the `gasPrice` different here?

Reply:
It's using the `fallBackPrice` prop in this case if an approval is needed: https://github.com/MetaMask/metamask-extension/blob/develop/test/jest/mock-store.js#L109